examples/api/http/http.go

```diff
@@ -0,0 +1,115 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"github.com/Southclaws/fault"
+	"github.com/Southclaws/fault/fctx"
+	"github.com/Southclaws/fault/fmsg"
+	"github.com/Southclaws/fault/ftag"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"go.uber.org/zap"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// Adding requestID in the fault context for logging purpose
+func RequestToFCtx(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		rid := middleware.GetReqID(ctx)
+		ctx = fctx.WithMeta(ctx, "request_id", rid)
+		ctx = fctx.WithMeta(ctx, "http_method", r.Method)
+		ctx = fctx.WithMeta(ctx, "request_path", r.URL.Path)
+		ctx = fctx.WithMeta(ctx, "remote_ip", r.RemoteAddr)
+		ctx = fctx.WithMeta(ctx, "protocol", r.Proto)
+		next.ServeHTTP(w, r.WithContext(ctx))
+	}
+	return http.HandlerFunc(fn)
+}
+
+// Adding any path variable in the fault context for logging purpose
+func PathVariableAsFCtx(pathVarName, fctxName string) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			if value := chi.URLParam(r, pathVarName); value != "" {
+				r = r.WithContext(fctx.WithMeta(r.Context(), fctxName, value))
+			}
+			next.ServeHTTP(w, r)
+		}
+		return http.HandlerFunc(fn)
+	}
+}
+
+func fctxToZapFields(ctx context.Context) []zap.Field {
+	faultFields := fctx.GetMeta(ctx)
+	fields := make([]zap.Field, 0, len(faultFields))
+	for k, v := range faultFields {
+		fields = append(fields, zap.String(k, v))
+	}
+	return fields
+}
+
+func LoggerRequest(logger *zap.Logger) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			ww := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
+			t1 := time.Now()
+			defer func() {
+				fields := fctxToZapFields(r.Context())
+				fields = append(fields, zap.Int("status", ww.Status()))
+				fields = append(fields, zap.Duration("latency", time.Since(t1)))
+				logger.Info("API Request", fields...)
+			}()
+			next.ServeHTTP(ww, r)
+		}
+		return http.HandlerFunc(fn)
+	}
+
+}
+
+func isInternalString(s string) bool {
+	return strings.HasPrefix(s, "<") && strings.HasSuffix(s, ">")
+}
+
+func toStackTrace(err error) string {
```

Comment:
is this mostly re-implementing what `func (f *container) Error() string` and `func (f *container) Format(s fmt.State, verb rune)` do?

Reply:
Yes, but I was aiming for a bit less noise, because using Format, I get
```
2023/09/09 11:12:52 ERROR db error: user id[321] not found
        examples/api/main.go:62
User not found
        examples/api/main.go:63
<ftag>
        examples/api/main.go:63
```
and with my custom function I get
```
2023/09/09 11:13:59 ERROR 
        db error: user id[321] not found
               examples/api/main.go:62
        User not found
               examples/api/main.go:63
```

The goal was to show you can display what you need (pick and choose). But like I said below, I don't like that we need to play with "isInternalString" to do so, so maybe we should just use format the the sake of the example.

Another option would be to change the Format method to "merge" the internal entries with the non-internal one, so that we get something like this
```
2023/09/09 11:13:59 ERROR 
        db error: user id[321] not found
               examples/api/main.go:62
        User not found [tags: notfound, xyz] /*(hypothetical if we have more than one tag)*/
               examples/api/main.go:63
```
but that is clearly out of scope for this PR.

Reply:
Should I remove this and use "Format" for the time being?

Reply:
I think yes, let's remove this and make the example as minimal as possible. Thanks!